pallets/omnipool/src/lib.rs

```diff
@@ -1275,14 +1283,16 @@ pub mod pallet {
 
 			Self::update_hdx_subpool_hub_asset(origin, state_changes.hdx_hub_amount)?;
 
-			Self::process_trade_fee(&who, asset_in, state_changes.fee.asset_fee)?;
+			Self::process_trade_fee(&who, asset_out, state_changes.fee.asset_fee)?;
```

Comment:
this looks like an important change, is this related to referrals?

Reply:
this is the part where we specify which one is the fee asset

since we used to calculate the amount in asset in, this had to chagne to be asset out as we fix it by this PR to be asset out.

and yes, it is used by referrals now.